Modify bale modal

dsusviela opened the pull request:
Check commit history for a complete run-down of files changed. Please note that several new files for testing have been created

Joaguirrem commented:
If I press on "Aceptar" button without having selected a material, the modal disappears without showing an alert. Same problem with the weight. I suggest adding a "Cancelar" button because back button doesn't work properly

dsusviela commented:
Fixed some errors related to the implementation

Joaguirrem commented:
Found the same errors of "Bale Edit Modal" in "Bale Create Modal"

dsusviela commented:
- Fixed several conflicts with master merge
- Fixed Modals not validating input
- Fixed Modals not closing when clicking outside of them
- Added back button for Modal closing
- Added strings for Modal validation

dsusviela commented:
- Fized CreateBaleModal being displayed by default

dsusviela commented:
- Removed a console log

andresk727 commented on src/components/Bale/BaleList.js:
```diff
@@ -38,28 +43,53 @@ class BaleList extends Component {
+                <PhoneBale
+                  id={item.id}
+                  onPressAction={this.props.openEditBaleModal}
+                />
```
You need to include type and weight in the PhoneBale props too, as the component needs them to show their info when clicked.

andresk727 commented:
![bale modal problem](https://user-images.githubusercontent.com/36371458/46980875-5e4f2e80-d0ac-11e8-8087-ee9c627e29eb.jpeg)
That error keeps appearing when closing and reopening the modal. It should be cleared when reopening.

dsusviela commented:
- Fixed state persisting when clicking away/back button to close modal
- Fixed props not being passed down from parent component "BaleList" to child "PhoneBale"

andresk727 commented:
![placeholder](https://user-images.githubusercontent.com/36371458/46982697-b6d5fa00-d0b3-11e8-9938-4e7d810e4947.jpeg)
Add a placeholder in the weight field.

dsusviela commented:
- Fixed weight placeholder not showing

andresk727 commented on src/components/Bale/styles.js:
```diff
@@ -124,6 +124,36 @@ const styles = StyleSheet.create({
+    borderColor: Colors.primary,
+    backgroundColor: Colors.white,
+    height: 300,
+    width: 400,
```
Can you use percentage instead?

dsusviela replied:
No. Since its the same modal for phones and tablets, theres no way to put percentages and get the size at a reasonable aesthetics with your suggestion

andresk727 commented on src/components/Bale/styles.js:
```diff
@@ -124,6 +124,36 @@ const styles = StyleSheet.create({
+  button: {
+    justifyContent: 'center',
+    height: 50,
+    width: 200,
```
Same here

andresk727 commented on src/components/Bale/styles.js:
```diff
@@ -124,6 +124,36 @@ const styles = StyleSheet.create({
+  },
+  modalTitleContainer: {
+    paddingTop: 15,
+    paddingBottom: 20,
```
And here

andresk727 commented on src/components/common/styles.js:
```diff
@@ -53,6 +53,36 @@ const styles = StyleSheet.create({
+  },
+  modalTitleContainer: {
+    paddingTop: 15,
+    paddingBottom: 20,
```
Use percentage if possible.

dsusviela commented on src/components/Bale/TabletBale.js:
```diff
@@ -35,12 +41,14 @@ TabletBale.propTypes = {
   type: '',
   weight: 0,
+  onPressAction: () => ({}),
 };
```
I will correct this but bear in mind this is an error that is in master currently and not a mistake on my part

dsusviela commented:
- Fixed typo on recylcablematerials variable
- Fixed code style 

Please note that modal styles are supposed to **not be percentages.** Since its the same modal for phones and tablets, theres no way to put percentages and get aesthetically reasonable sizes. We could have different modals for each device, but this solution still does not account different screen sizes.

dsusviela commented:
Modified:
 -Both localization/en.js and localization/es.js conflicts have been resolved

Please note that CircleCi tests are already broken on master branch and are currently being fixed at user-token-fix

dsusviela commented:
Fixed phonebale not showing the modal when info was toggled.
